zerver/lib/test_classes.py

```diff
@@ -1103,3 +1109,189 @@ def setUp(self) -> None:
 
     def setUpBeforeMigration(self, apps: StateApps) -> None:
         pass  # nocoverage
+
+
+class EmailChangeTestMixin(ZulipTestCase):
+    '''
+    Contains most of the email change tests and is subclassed in
+    test classes testing different auth flows.
+    A function `do_email_change` is defined in the subclasses to
+    authenticate based on the authentication flow.
+    '''
+    __unittest_skip__ = True
+
```

Comment:
this class should have a dummy `do_Change` function definition, to show the intended function signature, with a docstring quickly explaining its intent, and doing `raise NotImplementedError` because overriding it is obligatory

Reply:
Added, didn't know what the right thing to do to convey that overriding is obligatory so I initially just added a comment.